imap_processing/swapi/tests/test_swapi_l1.py

```diff
@@ -1 +1,211 @@
+
+
+def test_check_for_bad_data():
+    """Test for bad data"""
+    # create test data for this test
+    total_sweeps = 3
+    ds = xr.Dataset(
+        {
+            "PLAN_ID_SCIENCE": xr.DataArray(np.full((total_sweeps * 12), 1)),
+            "SWEEP_TABLE": xr.DataArray(np.repeat(np.arange(total_sweeps), 12)),
+            "MODE": xr.DataArray(np.full((total_sweeps * 12), 2)),
+        }
+    )
+
+    # Check for no bad data
+    bad_data_indices = check_for_bad_data(ds)
+    assert len(bad_data_indices) == 0
+
+    # Check for bad MODE data.
+    # This test returns this indices because MODE has 0, 1 values
+    # for the first two sweeps.
+    # TODO: update test when we update MODE from HVENG to HVSCI
+    ds["MODE"] = xr.DataArray(np.repeat(np.arange(total_sweeps), 12))
+    bad_data_indices = check_for_bad_data(ds)
+    assert np.all(bad_data_indices == np.arange(0, 24))
+
+    # Check for bad SWEEP_TABLE data.
+    # Reset MODE data and create first sweep to be mixed value
+    ds["MODE"] = xr.DataArray(np.full((total_sweeps * 12), 2))
+    ds["SWEEP_TABLE"][:12] = np.arange(0, 12)
+    assert np.all(check_for_bad_data(ds) == np.arange(0, 12))
+
+    # Check for bad PLAN_ID_SCIENCE data.
+    ds["SWEEP_TABLE"] = xr.DataArray(np.repeat(np.arange(total_sweeps), 12))
+    ds["PLAN_ID_SCIENCE"][24 : total_sweeps * 12] = np.arange(0, 12)
+    assert np.all(check_for_bad_data(ds) == np.arange(24, total_sweeps * 12))
+
+
+def test_decompress_count():
+    """Test for decompress count"""
+    # Test compressed and not overflow case
+    raw_value = np.array([[12]])
+    compression_flag = np.array([[1]])
+    expected_value = raw_value[0][0] * 16
+    returned_value = decompress_count(raw_value, compression_flag)
+    assert returned_value[0][0] == expected_value
+
+    # Test compressed and overflow case
+    raw_value = np.array([[0xFFFF]])
+    compression_flag = np.array([[1]])
+    expected_value = -1
+    returned_value = decompress_count(raw_value, compression_flag)
+    assert returned_value[0][0] == expected_value
+
+    # Test not compressed case
+    raw_value = np.array([[12]])
+    compression_flag = np.array([[0]])
+    expected_value = 12
+    returned_value = decompress_count(raw_value, compression_flag)
+    assert returned_value[0][0] == expected_value
+
+
+def test_find_sweep_starts():
+    """Test for find sweep starts"""
+    time = np.arange(26)
+    sequence_number = time % 12
+    ds = xr.Dataset({"SEQ_NUMBER": sequence_number}, coords={"Epoch": time})
+
+    start_indices = find_sweep_starts(ds)
+    assert np.all(start_indices == [0, 12])
+
+    ds["SEQ_NUMBER"].data[:12] = np.arange(3, 15)
+    start_indices = find_sweep_starts(ds)
+    assert np.all(start_indices == [12])
+
+    ds["SEQ_NUMBER"] = np.arange(3, 29)
+    start_indices = find_sweep_starts(ds)
+    assert np.all(start_indices == [])
+
+
+def test_get_full_indices():
+    """Test for correct full sweep indices"""
+    time = np.arange(14)
```

Comment:
Maybe test that you get two sweeps when expected? One is a bit easier to get right on accident.
```suggestion
    time = np.arange(26)
```